tools/driver/CMakeLists.txt

```diff
@@ -5,6 +5,8 @@
 # parser is built in.
 function(add_swift_parser_link_libraries target)
```

Comment:
Huh, I've missed this function in the past.

```
    target_link_libraries(${target}
                          PRIVATE swiftCore)
```

I have no idea how all the other tools are working, but we should fix this later (don't worry about it for this PR/Windows, we'll fix up on main after). This *should* be being added by `_add_swift_runtime_link_flags` from `add_swift_host_tool` but apparently isn't. Maybe autolinking is kicking in on Darwin?

Anyway, TLDR: `target_link_directories` is already done in `_add_swift_runtime_link_flags` already does `target_link_directories` so this change shouldn't be needed.

Reply:
It was needed as otherwise I ended up with references to libraries that are not available.

Reply:
I don't see how that could be, are you sure that wasn't before updating `_add_swift_runtime_link_flags`? `add_swift_host_tool` has:
```
  # Once the new Swift parser is linked in, every host tool has Swift modules.
  if (SWIFT_SWIFT_PARSER AND NOT ASHT_DOES_NOT_USE_SWIFT)
    set(ASHT_HAS_SWIFT_MODULES ON)
  endif()

  if (ASHT_HAS_SWIFT_MODULES)
    _add_swift_runtime_link_flags(${executable} "../lib" "${ASHT_BOOTSTRAPPING}")
  endif()
```

And you've added `target_link_libraries` there.